rio_cogeo/cogeo.py

```diff
@@ -564,17 +564,56 @@ def cog_validate(  # noqa: C901
                             )
                         )
 
-            block_offset = src.get_tag_item("BLOCK_OFFSET_0_0", "TIFF", bidx=1)
+            # Get blocks size
+            block_size = src.block_shapes[0]
+
+            # Extract number of blocks per row and column
+            blocks_per_row = src.width // block_size[1]
+            blocks_per_column = src.height // block_size[0]
+
+            # Initialize loop variables
+            y = 0
+            block_offset = None
+
+            # Find the first block with a valid block_offset
+            while y < blocks_per_column and block_offset is None:
+                x = 0
+                while x < blocks_per_row and block_offset is None:
+                    block_offset = src.get_tag_item(
+                        "BLOCK_OFFSET_%d_%d" % (x, y), "TIFF", bidx=1
+                    )
+                    x += 1
+                y += 1
```

Comment:
GDAL script is pretty close to this https://github.com/OSGeo/gdal/blob/master/swig/python/gdal-utils/osgeo_utils/samples/validate_cloud_optimized_geotiff.py#L78-L87 but I kinda prefer how they are doing it.

Reply:
> Do you also want it as an independent function

No that's fine